joblib/_parallel_backends.py

```diff
@@ -204,6 +207,7 @@ class SequentialBackend(ParallelBackendBase):
     """
 
     uses_threads = True
+    supports_timeout = False
     supports_asynchronous_callback = True
```

Comment:
Maybe put this at `False` so this is coherent.
Change test on line 802 for `parallel.py` to special case the `n_jobs=1` case.